### DISCUSSION TITLE
How to get request body from API call

### DISCUSSION THREAD

**Discussion by Enferum** [pocketbase/pocketbase#798] [Q&A]
I have take PocketBase as minimal PocketBase application.
Create a few collections and prepare API rules.
Now I need a change records with hook OnModelBeforeCreate but I didnt catch **how to get request body from API request**?
I'm looking at this example 
`app.OnModelBeforeCreate().Add(func(e *core.ModelEvent) error {
        log.Println(e.Model.TableName())
        return nil
    })`

**Comment by ganigeorgiev** [comment]
You can't access the request data in the `OnModel*` hooks. The `OnModel*` hooks are triggered whenever there is a `Dao` event (aka. db change).

Instead you can use the dedicated Request API hooks (usually with a suffix `Request`). You can access the request from there using `e.HttpContext`. For example:

```go
app.OnRecordBeforeCreateRequest().Add(func(e *core.RecordCreateEvent) error {
    log.Println(e.HttpContext) // echo.Context instance
    log.Println(e.Record)      // still unsaved so you can change the record data before being persisted
    
    return nil
})
```
You could find all hooks in https://pocketbase.io/docs/event-hooks/#record-hooks. _More detailed documentation and examples will be available soon._

**Reply by Enferum** [reply]
Many thanks for your help.
